pkg/search/postgres/common.go

```diff
@@ -1039,6 +1041,40 @@ func RunCursorQueryForSchema[T any, PT pgutils.Unmarshaler[T]](ctx context.Conte
 	}, closer, nil
 }
 
+// RunGetQueryForSchemaFn runs a query and call a callback for each row. It stops on first error.
+func RunGetQueryForSchemaFn[T any, PT pgutils.Unmarshaler[T]](ctx context.Context, schema *walker.Schema, q *v1.Query, db postgres.DB, callback func(obj PT) error) error {
+	if q == nil {
+		q = searchPkg.EmptyQuery()
+	}
+
+	query, err := standardizeQueryAndPopulatePath(ctx, q, schema, GET)
+	if err != nil {
+		return err
+	}
+	if query == nil {
+		return emptyQueryErr
+	}
+
+	queryStr := query.AsSQL()
+	rows, err := tracedQuery(ctx, db, queryStr, query.Data...)
+	if err != nil {
+		return err
+	}
+	defer rows.Close()
+
+	for rows.Next() {
```

Comment:
Should we handle `rows.Err()` after loop?

Reply:
I think we should.  Unless whatever `traceQuery` uses takes care of it.  Some of the pgx5 apis deal with it.  Some of the lower level ones do not.